test/sequential/sequencerStaking.cancel.test.ts

```diff
@@ -0,0 +1,403 @@
+  let chain: any;
+  let testUser1: User;
+  let testUser2: User;
+  let testUser2Address: string;
+  beforeEach(async () => {
+    chain = "Ethereum";
+    const notYetSequencer = await createACollatorUser();
+    testUser2 = await createACollatorUser();
+    const minToBeSequencer = await SequencerStaking.minimalStakeAmount();
+    await signTx(
+      await getApi(),
+      await SequencerStaking.provideSequencerStaking(
+        minToBeSequencer.addn(1000),
+        "Ethereum",
+      ),
+      notYetSequencer.keyRingPair,
+    ).then((events) => {
+      expectExtrinsicSucceed(events);
+    });
+    const sequencers = await SequencerStaking.activeSequencers();
+    expect(sequencers.toHuman().Ethereum).toContain(
+      notYetSequencer.keyRingPair.address,
+    );
+    await signTx(
+      await getApi(),
+      await SequencerStaking.provideSequencerStaking(
+        minToBeSequencer.addn(1000),
+        "Ethereum",
+      ),
+      testUser2.keyRingPair,
+    ).then((events) => {
+      expectExtrinsicSucceed(events);
+    });
+    testUser1 = notYetSequencer;
+    testUser2Address = testUser2.ethAddress.toString();
+    testUser1.addAsset(MGA_ASSET_ID);
+    testUser2.addAsset(MGA_ASSET_ID);
+  });
+  it("GIVEN a sequencer, WHEN <correctly> canceling an update THEN a % of the slash is given to it", async () => {
+    const { reqIdCanceled, api } = await createAnUpdateAndCancelIt(
+      testUser1,
+      testUser2,
+      chain,
+    );
+    await waitForNBlocks((await Rolldown.disputePeriodLength()).toNumber());
+    await Rolldown.waitForReadRights(testUser2Address, 50, "Ethereum");
+    const txIndex = await Rolldown.lastProcessedRequestOnL2(chain);
+    //we approve the cancellation
+    const cancelResolutionEvents = await Sudo.asSudoFinalized(
+      Sudo.sudoAsWithAddressString(
+        testUser2Address,
+        new L2Update(api)
+          .withCancelResolution(txIndex, reqIdCanceled, true)
+          .on("Ethereum")
+          .build(),
+      ),
+    );
+    await waitSudoOperationSuccess(cancelResolutionEvents, "SudoAsDone");
+    await testUser2.refreshAmounts(AssetWallet.BEFORE);
+    await testUser1.refreshAmounts(AssetWallet.BEFORE);
+    await waitForNBlocks((await Rolldown.disputePeriodLength()).toNumber());
+    await testUser2.refreshAmounts(AssetWallet.AFTER);
+    await testUser1.refreshAmounts(AssetWallet.AFTER);
+    const slashRewardCanceler = testUser2
+      .getAsset(MGA_ASSET_ID)
+      ?.amountAfter.free!.sub(
+        testUser2.getAsset(MGA_ASSET_ID)?.amountBefore.free!,
+      );
+    const slashFineUpdater = testUser1
+      .getAsset(MGA_ASSET_ID)
+      ?.amountBefore.reserved!.sub(
+        testUser1.getAsset(MGA_ASSET_ID)?.amountAfter.reserved!,
+      );
+    expect(slashRewardCanceler).bnGt(BN_ZERO);
+    expect(slashRewardCanceler).bnLt(await SequencerStaking.slashFineAmount());
+    expect(slashFineUpdater).bnEqual(await SequencerStaking.slashFineAmount());
```

Comment:
The amount the canceler gets is 20% of the slash fine amount I think. 
![image](https://github.com/user-attachments/assets/689478d5-f6f4-41cd-b4d6-729e988b22c1)
Plz adjust to be more precise

Reply:
Done